### ISSUE TITLE
Check typeof callback before invoking

### ISSUE THREAD

**Issue by janjongboom** [ARMmbed/mbed-connector-api-node#10]
Currently if you don't pass in a callback function your program will explode. That's not nice, sometimes I don't care about a callback.

```
TypeError: callback is not a function
    at /home/ubuntu/workspace/web/node_modules/mbed-connector/lib/subscriptions.js:82:7
    at Object.asyncCallbacks.(anonymous function) [as 225#bfc164a6-afea-4d72-9e34-73f6b854189c@fc5bae18-72d8-4d16-a0d0-00608fa76464/accelerometer/0/last_knock] (/home/ubuntu/workspace/web/node_modules/mbed-connector/lib/common.js:91:13)
    at EventEmitter.mbedConnector.asyncResponseHandler (/home/ubuntu/workspace/web/node_modules/mbed-connector/lib/notifications.js:238:44)
    at /home/ubuntu/workspace/web/node_modules/mbed-connector/lib/notifications.js:274:13
    at Array.forEach (native)
    at EventEmitter.mbedConnector.handleNotifications (/home/ubuntu/workspace/web/node_modules/mbed-connector/lib/notifications.js:273:32)
    at poll (/home/ubuntu/workspace/web/node_modules/mbed-connector/lib/notifications.js:164:15)
    at EventEmitter.mbedConnector.longPollRequestCallback (/home/ubuntu/workspace/web/node_modules/mbed-connector/lib/notifications.js:221:7)
    at Request._callback (/home/ubuntu/workspace/web/node_modules/mbed-connector/lib/common.js:78:21)
    at Request.self.callback (/home/ubuntu/workspace/web/node_modules/mbed-connector/node_modules/request/request.js:199:22)
```

**Comment by bridadan** [comment]
Ah yup, definitely need to be checking for this. Thanks for the catch.

**Comment by bridadan** [comment]
Fixed in 0ce1566b948c0d70febcf9c036ee7dba4e67e61d, added in release 2.0.0
